api/resourceserver/check_webhook.go

```diff
@@ -0,0 +1,70 @@
+package resourceserver
+
+import (
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/db"
+	"github.com/concourse/atc/dbng"
+	"github.com/tedsuo/rata"
+)
+
+// CheckResourceWebHook defines a handler for process a check resource request via an access token.
+func (s *Server) CheckResourceWebHook(pipelineDB db.PipelineDB, dbPipeline dbng.Pipeline) http.Handler {
+	logger := s.logger.Session("check-resource-webhook")
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resourceName := rata.Param(r, "resource_name")
+		accessToken := r.URL.Query().Get("access_token")
```

Comment:
Can we rename this to `webhook_token`? It has specific handling for this endpoint, whereas `access_token` sounds like a generic API for passing a token to all handlers.